seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java

```diff
@@ -0,0 +1,209 @@
+                                            .add("jobDag", logicalDag.getLogicalDagAsJson())
+                                            .add(
+                                                    "pluginJarsUrls",
+                                                    (JsonValue)
+                                                            jobImmutableInformation
+                                                                    .getPluginJarsUrls().stream()
+                                                                    .map(
+                                                                            url -> {
+                                                                                JsonObject jarUrl =
+                                                                                        new JsonObject();
+                                                                                jarUrl.add(
+                                                                                        "jarPath",
+                                                                                        url
+                                                                                                .toString());
+                                                                                return jarUrl;
+                                                                            })
+                                                                    .collect(
+                                                                            JsonArray::new,
+                                                                            JsonArray::add,
+                                                                            JsonArray::add))
+                                            .add(
+                                                    "isStartWithSavePoint",
+                                                    jobImmutableInformation.isStartWithSavePoint())
+                                            .add(
+                                                    "metrics",
+                                                    JsonUtil.toJsonObject(
+                                                            getJobMetrics(jobMetrics)));
+                                })
+                        .collect(JsonArray::new, JsonArray::add, JsonArray::add);
+        this.prepareResponse(command, jobs);
+    }
+
+    private Map<String, Long> getJobMetrics(String jobMetrics) {
+        Map<String, Long> metricsMap = new HashMap<>();
+        long sourceReadCount = 0L;
+        long sinkWriteCount = 0L;
+        try {
+            JsonNode jobMetricsStr = new ObjectMapper().readTree(jobMetrics);
+            for (int i = 0; i < jobMetricsStr.get(SOURCE_RECEIVED_COUNT).size(); i++) {
+                JsonNode sourceReader = jobMetricsStr.get(SOURCE_RECEIVED_COUNT).get(i);
+                JsonNode sinkWriter = jobMetricsStr.get(SINK_WRITE_COUNT).get(i);
```

Comment:
`jobMetricsStr.get(SOURCE_RECEIVED_COUNT)` and `jobMetricsStr.get(SINK_WRITE_COUNT)` can be placed outside the for loop